services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jIslRepository.java

```diff
@@ -0,0 +1,88 @@
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories.impl;
+
+import org.openkilda.model.Isl;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.repositories.IslRepository;
+
+import org.neo4j.ogm.cypher.query.SortOrder;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Neo4J OGM implementation of {@link IslRepository}.
+ */
+public class Neo4jIslRepository extends Neo4jGenericRepository<Isl> implements IslRepository {
+    public Neo4jIslRepository(Neo4jSessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public Isl findByEndpoint(SwitchId switchId, int port) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("src_switch", switchId.toString());
+        parameters.put("src_port", port);
+
+        return getSession().queryForObject(Isl.class, "MATCH (src:switch)-[target:isl]->(:switch)\n"
+                + " WHERE src.name=$src_switch AND target.src_port=$src_port\n"
+                + " RETURN target", parameters);
+    }
+
+    @Override
+    public Iterable<Isl> findOccupiedByFlow(String flowId, boolean ignoreBandwidth, long requiredBandwidth) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("flow_id", flowId);
+        parameters.put("ignore_bandwidth", ignoreBandwidth);
+        parameters.put("requested_bandwidth", requiredBandwidth);
+
+        String query = "MATCH (src:switch)-[fs:flow_segment{flowid: $flow_id}]->(dst:switch) "
+                + "MATCH (src)-[link:isl]->(dst) "
+                + "WHERE src.state = 'active' AND dst.state = 'active' AND link.status = 'active' "
+                + " AND link.src_port = fs.src_port AND link.dst_port = fs.dst_port "
+                + " AND ($ignore_bandwidth OR link.available_bandwidth + fs.bandwidth >= $requested_bandwidth) "
+                + "RETURN src, link, dst";
+
+        return getSession().query(Isl.class, query, parameters);
+    }
+
+    @Override
+    public Iterable<Isl> findActiveWithAvailableBandwidth(boolean ignoreBandwidth, long requiredBandwidth) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("ignore_bandwidth", ignoreBandwidth);
+        parameters.put("requested_bandwidth", requiredBandwidth);
+
+        String query = "MATCH (src:switch)-[link:isl]->(dst:switch) "
+                + "WHERE src.state = 'active' AND dst.state = 'active' AND link.status = 'active' "
+                + " AND src.name IS NOT NULL AND dst.name IS NOT NULL "
+                + " AND ($ignore_bandwidth OR link.available_bandwidth >= $requested_bandwidth) "
```

Comment:
`ignore_bandwidth` can(or even should) be handled by caller... by sending 0 in `requiredBandwidth`.

Reply:
Fixed.